test/java/us/klette/constantstring/CStringSubRangeTest.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2014-2015, Kristian Klette
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the Qulice.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package us.klette.constantstring;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CStringSubRangeTest {
+
+    @Test
+    public void testToString() throws Exception {
+        CStringSubRange foobar = new CStringSubRange(CString.create("foobar"), 3, 4);
+        String s = foobar.toString();
+        assertThat(s).isEqualTo("foobar".substring(3, 4));
+    }
```

Comment:
I really dislike this one:

* Name - no reference to actual business requirement
* "foobar" is name with 0 domain knowledge value, as is "s"
* I do not understand why you picked the indices 3 and 4, what is their relation to the value and expected result
* Also, wouldn't it be better to use the public API of the lib, i.e. create a CString and call substring on it? Thus you're free to rename your classes and lot other without the test becoming outdated; the abstraction level of the test becomes higher, closer to business
(I wanted to propose a better name but I really fail to understand why substring(.., 3,4) should return the string unchanged)